node/pkg/governor/governor_db.go

```diff
@@ -233,5 +242,27 @@ func (gov *ChainGovernor) reloadTransfer(xfer *db.Transfer) error {
 		return err
 	}
 	ce.transfers = append(ce.transfers, transfer)
+
+	// Reload flow-cancel transfers for the TargetChain. This is important when node restarts so that a corresponding,
+	// inverse transfer is added to the TargetChain. This is already done during the `ProcessMsgForTime` loop but
+	// that function does not capture flow-cancelling when the node is restarted.
+	tokenEntry := gov.tokens[tk]
+	if tokenEntry != nil {
```

Comment:
To me, I'd prefer that the *negative* part of the flow were stored in the database. However, the dBTransfer object is unsigned, meaning that's not possible without modifying the DB entries. So, adding in the flow canceling upon reload makes sense.

One issue this may have is a race condition on token entries on reload vs. what was originally set at the time of usage. We are *assuming* that all of the token entries in the flow cancel list are the same now as prior to the reload, which may not be the case.

So, a token that was NOT flow canceled before will be on reload and a token that wasn't originally flow canceled WILL be on reload. I don't see how to avoid this in the current architecture though... Hmmm...

Reply:
After discussions in Slack, we decided that the complexity of fixing this issue is too high in a nice way. It would require modifying the database storage of transfers to include an unsigned value, which would be difficult to get right in all of the places. 

If the second case (token removed) is triggered in production, then the governor just panics and doesn't allow anything through. This can also happen through governor limit decreases midcycle and reload. 

A better solution to this is changing the error handling on the reload to allow for the sum to be larger than dailylimit on reload. Although there may be a delay on the chain with the removed flow cancel entries, it's very unlikely to happen. 

In the backfilled flow cancels (token added case), it's fine. We've decided to add the token in so it's not much of a threat.